firedrake/slate/slac/compiler.py

```diff
@@ -36,17 +40,29 @@
 
 PETSC_DIR = get_petsc_dir()
 
+supported_integral_types = [
+    "cell",
+    "interior_facet",
+    "exterior_facet",
+    "interior_facet_horiz",
+    "interior_facet_vert",
+    "exterior_facet_top",
+    "exterior_facet_bottom",
+    "exterior_facet_vert"
+]
+
 
 def compile_expression(slate_expr, tsfc_parameters=None):
     """Takes a SLATE expression `slate_expr` and returns the appropriate
     :class:`firedrake.op2.Kernel` object representing the SLATE expression.
 
     :arg slate_expr: a :class:'TensorBase' expression.
     :arg tsfc_parameters: an optional `dict` of form compiler parameters to
-                          be passed onto TSFC during the compilation of ufl forms.
+                          be passed onto TSFC during the compilation of
+                          ufl forms.
     """
     if not isinstance(slate_expr, TensorBase):
-        raise ValueError("Expecting a `slate.TensorBase` expression, not a %r" % slate_expr)
+        raise ValueError("Expecting a `TensorBase` expr, not %r" % slate_expr)
 
```

Comment:
I think the plan here was to remember the compiled kernels on the `TensorBase` object, so that if we see it again, we don't have to go through this long symbolic compilation step.

That would look like this:

```python

class TensorBase(...):

    def __init__(self, ...):
          self._kernels = None
   ...

def compile_expression(expr, ...):
    if not isinstance(expr, TensorBase):
        raise ValueError(...)
    if expr._kernels is not None:
        return expr._kernels
    ...

   kernels = (SplitKernel(idx, kinfo),)
   expr._kernels = kernels
   return kernels
```